src/mesido/potential_errors.py

```diff
@@ -0,0 +1,65 @@
+from typing import Dict, List
+
+from mesido.exceptions import MesidoAssetIssue
+from mesido.network_common import MesidoAssetIssueType
+
+AssetId = str
+ErrorMessage = str
+
+
+class PotentialErrors:
+    """Singleton, do not instantiate. Use POTENTIAL_ERRORS."""
+
+    _gathered_potential_issues: Dict[MesidoAssetIssueType, Dict[AssetId, List[ErrorMessage]]]
+
+    def __init__(self):
+        self._gathered_potential_issues = {}
+
+    def add_potential_issue(
+        self, issue_type: MesidoAssetIssueType, asset_id: AssetId, error_message: ErrorMessage
+    ) -> None:
+        """
+        Add potential issues to _gathered_potential_issues.
+
+        """
+        self._gathered_potential_issues.setdefault(issue_type, {}).setdefault(asset_id, "")
+        self._gathered_potential_issues[issue_type][asset_id] = error_message
+
+    def have_issues_for(self, issue_types: List[MesidoAssetIssueType]) -> Dict[
+        MesidoAssetIssueType,
+        bool,
+    ]:
+        """
+        Check if the potential issue exists and mark them as True for the specific issue type.
+
+        """
+        result = {}
+        for ii in range(len(issue_types)):
+            if (
+                issue_types[ii] in self._gathered_potential_issues
+                and len(self._gathered_potential_issues) > 0
```

Comment:
why do you also check if the dictionary has a length? It should be a dictionary so no need to check a length, or do you want to check the length of a nested dict?

Reply:
Good point. This is due too a previous variation of the code. Len() check has been removed.

Reply:
This is due to a previous variation of the variables. The len() check has been removed